modules/Bio/EnsEMBL/Feature.pm

```diff
@@ -1445,6 +1445,49 @@ sub get_nearest_Gene {
   }
 }
 
+=head2 feature_so_acc
+
+  Description: This method returns a string containing the SO accession number of the feature
+  Returntype : String (Sequence Ontology accession number)
+  Exceptions : Thrown if no SO accession can be returned for caller object type
+
+=cut
+
+sub feature_so_acc {
+  my ($self) = @_;
+
+  # Classes that inherit Feature will inherit this method
+  # Slice and UTR override this method for better SO accs based on type
+  my %feature_so_mapping = (
```

Comment:
All static lists should be class properties, not dynamically defined. It's faster and saves memory reallocation.

I would have preferred to have this list of SO terms dispersed into the relevant classes as package level "constants" so that each team may maintain their own SO term assignments, but I expect the support path leads back to us either way if a pipeline breaks, so it perhaps does not matter. The exceptional behaviour is still needed for non-Features, but it would have been nice to have a more OO-approach.

Reply:
Have ended up refactoring this feature to explicit feature_so_acc methods in the actual classes. If required and not overridden, then will throw an error.
Also took the static lists to class properties where applicable.

Reply:
When feature has the ok will submit PR to Compara, FuncGen and Variation with the missing class updates.
    'Bio::EnsEMBL::Variation::VariationFeature'  => 'SO:0001060', # sequence variant
    'Bio::EnsEMBL::Variation::StructuralVariationFeature' => 'SO:0001537', # structural variant
    'Bio::EnsEMBL::Compara::ConstrainedElement' => 'SO:0001009', # DNA_constraint_sequence ????
    'Bio::EnsEMBL::Funcgen::RegulatoryFeature' => 'SO:0005836', # regulatory_region